lib/Web/Machine/FSM/States.pm

```diff
@@ -65,10 +65,15 @@ sub _ensure_quoted_header {
 
 sub _get_acceptable_content_type_handler {
     my ($resource, $request) = @_;
-    my $acceptable = match_acceptable_media_type(
-        ($request->header('Content-Type') || 'application/octet-stream'),
-        $resource->content_types_accepted
-    );
+    my $metadata = _metadata($request);
+    my $acceptable = $metadata->{'acceptable_request_handler'};
+    if ( not defined $acceptable ) {
+        $acceptable = match_acceptable_media_type(
+            ($request->header('Content-Type') || 'application/octet-stream'),
+            $resource->content_types_accepted
+        );
+        $metadata->{'acceptable_request_handler'} = $acceptable;
```

Comment:
> I think you want to store the return val of pair_value($acceptable), not the pair, which is what you're doing now.

Right. Changed the patch to reflect this:

https://github.com/smithfarm/webmachine-perl/commit/5cdef1c5692d244fc78ffd43b6d5d2da51f3c0d3